sycl/include/CL/sycl/accessor.hpp

```diff
@@ -354,23 +365,65 @@ class image_accessor
                  handler &CommandGroupHandlerRef, int ImageElementSize)
 #ifdef __SYCL_DEVICE_ONLY__
   {
-    // TODO: Implement this function.
+    // No implementation needed for device. The constructor is only called by
+    // host.
   }
 #else
       : AccessorBaseHost(id<3>(0, 0, 0) /* Offset,*/,
                          detail::convertToArrayOfN<3, 1>(ImageRef.get_range()),
                          detail::convertToArrayOfN<3, 1>(ImageRef.get_range()),
                          AccessMode, detail::getSyclObjImpl(ImageRef).get(),
                          Dimensions, ImageElementSize),
-        MImageSize(ImageRef.get_size()), MImageCount(ImageRef.get_count()) {
+        MImageCount(ImageRef.get_count()),
+        MImageSize(MImageCount * ImageElementSize) {
     // TODO: Implement this function.
   }
 #endif
 
-  /* TODO -- common interface members -- */
+  /* -- common interface members -- */
+
+  // operator == and != need to be defined for host application only as per the
+  // SYCL spec 1.2.1
+  bool operator==(const image_accessor &Rhs) const {
+#ifdef __SYCL_DEVICE_ONLY
+    // The __SYCL_DEVICE_ONLY need to be defined for host application with
+    // device compiler.
+    return MImageObj == Rhs.MImageObj;
```

Comment:
Not sure it will compile if you use such code for device.

Reply:
removed the definition for the operator for __SYCL_DEVICE_ONLY__